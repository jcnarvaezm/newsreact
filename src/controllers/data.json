{
  "status": "ok",
  "totalResults": 70,
  "articles": [
    {
      "source": { "id": null, "name": "CNBC" },
      "author": "Lauren Thomas",
      "title": "The biggest US mall owner, Simon Property, sues Gap over skipped rent payments - CNBC",
      "description": "Simon Property Group said Gap has failed to pay more than $65.9 million in rent and other charges during the coronavirus pandemic.",
      "url": "https://www.cnbc.com/2020/06/04/the-biggest-us-mall-owner-simon-property-sues-gap-over-skipped-rent.html",
      "urlToImage": "https://image.cnbcfm.com/api/v1/image/106532376-15892353373142018-12-11t210429z_1042735456_rc148ec22990_rtrmadp_3_usa-holidayshopping.jpg?v=1591271036",
      "publishedAt": "2020-06-04T12:41:16Z",
      "content": "The biggest mall owner in the country ,Simon Property Group, is suing one of its biggest tenants, Gap, saying it failed to pay more than $65.9 million in rent and other charges due during the coronav… [+1698 chars]"
    },
    {
      "source": { "id": null, "name": "CNBC" },
      "author": "Silvia Amaro",
      "title": "European Central Bank takes its pandemic bond buying to 1.35 trillion euros to try and prop up economy - CNBC",
      "description": "The ECB announced that it will increase its Pandemic Emergency Purchase Programme by 600 billion euros ($672 billion), as it attempts to bolster the region's economy following the coronavirus crisis.",
      "url": "https://www.cnbc.com/2020/06/04/european-central-bank-ramps-up-its-pandemic-bond-buying-to-1point35-trillion-euros.html",
      "urlToImage": "https://image.cnbcfm.com/api/v1/image/106354120-1579884106653rts2zk7p.jpg?v=1579884172",
      "publishedAt": "2020-06-04T12:37:48Z",
      "content": "The European Central Bank (ECB) announced Thursday that it will increase its Pandemic Emergency Purchase Programme (PEPP) by 600 billion euros ($672 billion), as it attempts to bolster the region's e… [+2525 chars]"
    },
    {
      "source": { "id": null, "name": "CNBC" },
      "author": "Peter Schacknow",
      "title": "Stocks making the biggest moves in the premarket: JM Smucker, Tiffany, Navistar, Costco & more - CNBC",
      "description": "The stocks making the biggest moves in premarket trading include Tiffany, J.M. Smucker, Navistar, Costco, and more.",
      "url": "https://www.cnbc.com/2020/06/04/stocks-making-the-biggest-moves-in-the-premarket-jm-smucker-tiffany-navistar-costco-more.html",
      "urlToImage": "https://image.cnbcfm.com/api/v1/image/104568946-GettyImages-450109704.jpg?v=1555584407",
      "publishedAt": "2020-06-04T12:08:19Z",
      "content": "Take a look at some of the biggest movers in the premarket:\r\nJ.M. Smucker (SJM) The food producer reported quarterly earnings of $2.57 per share, beating the consensus estimate of $2.29 a share. Reve… [+3205 chars]"
    },
    {
      "source": { "id": "reuters", "name": "Reuters" },
      "author": "Aditya Kalra",
      "title": "Exclusive: Amazon in talks to buy $2 billion stake in Indian telco Bharti Airtel - sources - Reuters",
      "description": "Amazon.com is in early-stage talks to buy a stake worth at least $2 billion in Indian mobile operator Bharti Airtel , three people with knowledge of the discussions told Reuters, in a move that could turbocharge India's digital economy.",
      "url": "https://www.reuters.com/article/us-amazon-india-bharti-exclusive-idUSKBN23B1F8",
      "urlToImage": "https://s4.reutersmedia.net/resources/r/?m=02&d=20200604&t=2&i=1521061419&w=1200&r=LYNXMPEG530WS",
      "publishedAt": "2020-06-04T12:07:17Z",
      "content": "NEW DELHI/HONG KONG (Reuters) - Amazon.com (AMZN.O) is in early-stage talks to buy a stake worth at least $2 billion in Indian mobile operator Bharti Airtel (BRTI.NS), three people with knowledge of … [+3228 chars]"
    },
    {
      "source": { "id": null, "name": "NPR" },
      "author": "",
      "title": "'I Didn't Deserve It': Pandemic Shut Down His Barbershop, Then A Fire Destroyed It - NPR",
      "description": "Trevon Ellis spent years building up his Minneapolis barbershop, luring customers with smart haircuts, snacks and friendly conversation. It took just one terrible night to destroy it all.",
      "url": "https://www.npr.org/2020/06/04/868972077/i-didn-t-deserve-it-pandemic-shut-down-his-barbershop-then-a-fire-destroyed-it",
      "urlToImage": "https://media.npr.org/assets/img/2020/06/03/barberfirefinal_barrayn_npr014_wide-eae872649ae95f992616550c1ee27bf4345737c3.jpg?s=1400",
      "publishedAt": "2020-06-04T12:05:44Z",
      "content": "Trevon Ellis, owner of the Fade Factory, stands in front of the building where his barbershop was destroyed in Minneapolis. Sometime late Saturday someone broke into the shop and lit it on fire.\r\nLay… [+3591 chars]"
    },
    {
      "source": { "id": null, "name": "Journal of Accountancy" },
      "author": "By Jeff Drew",
      "title": "PPP forgiveness changes coming as Senate passes House bill - Journal of Accountancy",
      "description": "The Senate passed a House bill that gives small businesses and other entities that received loans from the Paycheck Protection Program 16 additional weeks to spend the funds and qualify for full forgiveness of the loans.",
      "url": "https://www.journalofaccountancy.com/news/2020/jun/ppp-loan-forgiveness-changes-coming.html",
      "urlToImage": "https://www.journalofaccountancy.com/content/dam/jofa/news/small-biz-planning-640.jpg",
      "publishedAt": "2020-06-04T12:01:50Z",
      "content": "The U.S. Senate passed the House version of Paycheck Protection Program (PPP) legislation Wednesday night, tripling the time allotted for small businesses and other PPP loan recipients to spend the f… [+4296 chars]"
    },
    {
      "source": { "id": null, "name": "CNBC" },
      "author": "Phil LeBeau",
      "title": "American Airlines and other carriers are adding summer flights as passengers slowly return - CNBC",
      "description": "American Airlines says it plans to fly 55% of its domestic schedule in July, up dramatically from May when it flew 20% of its schedule from a year earlier.",
      "url": "https://www.cnbc.com/2020/06/04/american-airlines-and-other-carriers-are-adding-summer-flights-as-passengers-slowly-return.html",
      "urlToImage": "https://image.cnbcfm.com/api/v1/image/106549052-1590155652068gettyimages-1214277463.jpeg?v=1590155702",
      "publishedAt": "2020-06-04T11:28:14Z",
      "content": "Slowly, the airline business is coming back to life as airlines set schedules for July that include more flights. \r\nThe latest is American Airlines, which says it plans to fly 55% of its domestic sch… [+2299 chars]"
    },
    {
      "source": { "id": "reuters", "name": "Reuters" },
      "author": "Ben Klayman",
      "title": "Exclusive: GM plans electric van for business users in bid to pre-empt Tesla - Reuters",
      "description": "General Motors Co is developing an electric van aimed at business users, joining a growing list of carmakers planning EVs for the same segment which includes customers such as Amazon.com Inc and United Parcel Service Inc, five people familiar with the plans t…",
      "url": "https://www.reuters.com/article/us-autos-electric-vans-exclusive-idUSKBN23B1G1",
      "urlToImage": "https://s4.reutersmedia.net/resources/r/?m=02&d=20200604&t=2&i=1521063332&w=1200&r=LYNXMPEG530XZ",
      "publishedAt": "2020-06-04T10:46:42Z",
      "content": "DETROIT (Reuters) - General Motors Co is developing an electric van aimed at business users, joining a growing list of carmakers planning EVs for the same segment which includes customers such as Ama… [+4239 chars]"
    },
    {
      "source": { "id": null, "name": "CNBC" },
      "author": "Jeff Cox",
      "title": "'The Covid-19 recession is over,' says economist Zandi as May job losses not as bad as feared - CNBC",
      "description": "Finding good news in the loss of another 3 million jobs isn't easy, but the steep drop in private payrolls for May could be a sign that the worst over.",
      "url": "https://www.cnbc.com/2020/06/03/the-covid-19-recession-is-over-says-economist-zandi-as-may-job-losses-not-as-bad-as-feared.html",
      "urlToImage": "https://image.cnbcfm.com/api/v1/image/106557747-1590789318036gettyimages-1216198284.jpeg?v=1590789403",
      "publishedAt": "2020-06-04T10:43:03Z",
      "content": "Finding good news in the loss of another 3 million jobs isn't easy, but the steep drop in private payrolls for May could be a sign that the worst over.\r\nADP reported Wednesday that companies shed ano… [+2546 chars]"
    },
    {
      "source": { "id": null, "name": "NorthJersey.com" },
      "author": "John Connolly",
      "title": "$136.4 million Powerball winning ticket sold in West Virginia - NorthJersey.com",
      "description": "One ticket sold in West Virginia for Wednesday's drawing matched all six numbers to win the $136.4 million jackpot,",
      "url": "https://www.northjersey.com/story/news/nation/2020/06/04/powerball-136-4-million-winning-ticket-sold-west-virginia/3142013001/",
      "urlToImage": "https://www.gannett-cdn.com/presto/2019/01/02/PCHH/20d92914-a71c-429c-9d5e-d992b7b23aa0-CHLBrd_04-23-2017_Daily_1_A001__2017_04_22_IMG_powerball.jpg_1_1_8AI3LBCM_L1012980066_IMG_powerball.jpg_1_1_8AI3LBCM.jpg?crop=999,563,x0,y0&width=1600&height=800&fit=bounds",
      "publishedAt": "2020-06-04T10:17:43Z",
      "content": "A lucky lottery player in West Virginia has millions of reasons to smile. \r\nOnly one ticket sold for Wednesday's drawing matched all six numbers to win the $136.4 million jackpot, according to the Po… [+999 chars]"
    },
    {
      "source": { "id": null, "name": "CNBC" },
      "author": "Fred Imbert",
      "title": "Stock futures drop as Wall Street rally looks set to pause - MSN Money",
      "description": "Futures tied to the major U.S. indexes were lower in early trading Thursday on the heels of yet another robust day on Wall Street.",
      "url": "https://www.cnbc.com/2020/06/03/stock-market-futures-open-to-close-news.html",
      "urlToImage": "https://image.cnbcfm.com/api/v1/image/106507926-1588006171485rtx7ftw9.jpg?v=1590097529",
      "publishedAt": "2020-06-04T09:30:00Z",
      "content": "Futures contracts tied to the major U.S. stock indexes fell on Thursday as Wall Street takes a breath following a blistering start to June. \r\nDow Jones Industrial Average futures traded 109 points lo… [+2415 chars]"
    },
    {
      "source": { "id": null, "name": "MarketWatch" },
      "author": "Steve Goldstein",
      "title": "ZoomInfo prices IPO above already-raised target - MarketWatch",
      "description": "",
      "url": "http://www.marketwatch.com/story/zoominfo-prices-ipo-above-already-raised-target-2020-06-04",
      "urlToImage": "https://s.wsj.net/public/resources/MWimages/MW-GP644_MicroS_ZG_20180906154215.jpg",
      "publishedAt": "2020-06-04T08:46:00Z",
      "content": "ZoomInfo Technologies Inc.\r\n ZI, \r\nsaid it's priced its initial public offering of 44.5 million shares at $21. That pricing tops an already-hiked range of $19 to $20. ZoomInfo said it will receive ne… [+95 chars]"
    },
    {
      "source": { "id": "techcrunch", "name": "TechCrunch" },
      "author": "Catherine Shu",
      "title": "Amazon Air adds 12 new aircraft to its cargo fleet, expands its ground operations - TechCrunch",
      "description": "Amazon announced today it has added 12 new cargo aircraft to Amazon Air, bringing its total fleet to more than 80 aircraft, in part because of increased demand for shipments during the COVID-19 pandemic. The converted Boeing 767-300 were leased from the Air T…",
      "url": "http://techcrunch.com/2020/06/04/amazon-air-adds-12-new-aircraft-to-its-cargo-fleet-expands-its-ground-operations/",
      "urlToImage": "https://techcrunch.com/wp-content/uploads/2020/06/GettyImages-586217458.jpg?w=585",
      "publishedAt": "2020-06-04T07:53:03Z",
      "content": "Amazon announced today it has added 12 new cargo aircraft to Amazon Air, bringing its total fleet to more than 80 aircraft, in part because of increased demand for shipments during the COVID-19 pande… [+1882 chars]"
    },
    {
      "source": { "id": null, "name": "MarketWatch" },
      "author": "Pietro Lombardi",
      "title": "LVMH says not considering buying Tiffany stock - MarketWatch",
      "description": "",
      "url": "https://www.marketwatch.com/story/lvmh-says-not-considering-buying-tiffany-stock-2020-06-04",
      "urlToImage": "https://mw3.wsj.net/mw5/content/logos/mw_logo_social.png",
      "publishedAt": "2020-06-04T06:58:00Z",
      "content": "Read Next\r\nWarner Music Group Corp. priced its initial public offering on Wednesday at $25 a share, the higher end of its $23 to $26 price range, and increased the size of the deal to 77 million shar… [+839 chars]"
    },
    {
      "source": { "id": null, "name": "Fox Business" },
      "author": "Associated Press",
      "title": "Nevada betting on health safety as Las Vegas casinos reopen - Fox Business",
      "description": "Las Vegas casinos will be opening their doors for the first time in 78 days.",
      "url": "https://www.foxbusiness.com/markets/nevada-betting-on-health-safety-as-las-vegas-casinos-reopen",
      "urlToImage": "https://a57.foxnews.com/static.foxbusiness.com/foxbusiness.com/content/uploads/2020/06/0/0/ba95bb6f-wire01.jpeg?ve=1&tl=1",
      "publishedAt": "2020-06-04T06:17:45Z",
      "content": "After 78 days of historic quiet, cards will be cut, dice will roll and jackpots can jingle again 12:01 a.m. Thursday at casinos in Las Vegas and throughout Nevada.\r\nThere will be big splashes — even … [+4489 chars]"
    },
    {
      "source": { "id": null, "name": "Fox Business" },
      "author": "Reuters",
      "title": "US new weekly jobless claims seen falling below 2 million - Fox Business",
      "description": "Thursday's report would suggest the worst is over for the labor market",
      "url": "https://www.foxbusiness.com/markets/us-new-weekly-jobless-claims-seen-falling-below-2-million",
      "urlToImage": "https://a57.foxnews.com/static.foxbusiness.com/foxbusiness.com/content/uploads/2020/05/0/0/unemployment-iStock-93838.jpg?ve=1&tl=1",
      "publishedAt": "2020-06-04T06:01:32Z",
      "content": "The number of Americans filing for unemployment benefits likely dropped below 2 million last week for the first time since mid-March, but remains astonishingly high as companies adjust to an environm… [+4836 chars]"
    },
    {
      "source": { "id": null, "name": "Cointelegraph" },
      "author": "Turner Wright",
      "title": "‘No Message Was Signed’: Craig Wright Refutes Tulip Trust Fabrication - Cointelegraph",
      "description": "Craig Wright has refuted the notion he doesn’t have the private keys to the ‘Satoshi’ addresses that he claimed in court as his own",
      "url": "https://cointelegraph.com/news/no-message-was-signed-craig-wright-refutes-tulip-trust-fabrication",
      "urlToImage": "https://s3.cointelegraph.com/storage/uploads/view/1af5956522ce90d117cfd3ccf6a8a6ec.jpg",
      "publishedAt": "2020-06-04T04:33:20Z",
      "content": "Craig Wright maintains that he alone has access to the ‘Satoshi’ Bitcoin addresses filed in court, despite a message apparently signed by 145 of the addresses calling him “a liar and a fraud”.\r\nWrigh… [+2470 chars]"
    },
    {
      "source": { "id": null, "name": "WTAE Pittsburgh" },
      "author": null,
      "title": "Rivers Casino to reopen next Tuesday, masks will be required - WTAE Pittsburgh",
      "description": "To comply with state guidelines and promote social distancing, Rivers Casino will be limiting restaurants and gaming floor capacity to 50% occupancy until further notice.",
      "url": "https://www.wtae.com/article/rivers-casino-to-reopen-next-tuesday-masks-will-be-required/32759461",
      "urlToImage": "https://kubrick.htvapps.com/htv-prod-media.s3.amazonaws.com/images/rivers-casino-2-1563487227.jpg?crop=1.00xw:1.00xh;0,0&resize=1200:*",
      "publishedAt": "2020-06-04T04:09:37Z",
      "content": "PITTSBURGH —Rivers Casino Pittsburgh announced Wednesday that it will reopen on Tuesday, June 9, at 9 a.m. \r\nRivers Casino temporarily suspended operations on March 15, ahead of the governors orders,… [+2331 chars]"
    },
    {
      "source": { "id": null, "name": "Vulture" },
      "author": "Justin Curto",
      "title": "AMC Theatres Has ‘Substantial Doubt’ It Can Reopen - Vulture",
      "description": "The company reported it is “generating effectively no revenue.”",
      "url": "http://www.vulture.com/2020/06/amc-theatres-doubt-coronavirus-reopening.html",
      "urlToImage": "https://pyxis.nymag.com/v1/imgs/43b/98f/4db1395deceb3bcb494bcbed353d440766-amc--.1x.rsocial.w1200.jpg",
      "publishedAt": "2020-06-04T03:25:00Z",
      "content": "When the coronavirus pandemic caused entertainment venues to close, small, independent theaters looked to be in the most danger. Now, with venues in most states still closed nearly three months in, t… [+1483 chars]"
    },
    {
      "source": { "id": null, "name": "Hawaii News Now" },
      "author": "HNN Staff",
      "title": "Governor declines to set a date for reopening Hawaii tourism - Hawaii News Now",
      "description": "In a candid, one-on-one interview with Hawaii News Now, Ige talked about why he’s holding off on setting any dates for rebooting tourism and why he expects foreign travelers may be coming here first.",
      "url": "https://www.hawaiinewsnow.com/2020/06/03/governor-declines-set-date-when-tourism-will-restart-hawaii/",
      "urlToImage": "https://www.hawaiinewsnow.com/resizer/Hz-DaFDwO7FXbMFFHIYcC45JDgE=/1200x0/cloudfront-us-east-1.images.arcpublishing.com/raycom/34KRE5663VF2PGIV4H2WXPPAKI.jpg",
      "publishedAt": "2020-06-04T03:12:00Z",
      "content": "HONOLULU, Hawaii (HawaiiNewsNow) - Gov. David Ige declined to offer any specific timeline Wednesday for when tourism would be rebooted in Hawaii but said he has a plan to do it safely.\r\nIn a candid, … [+1956 chars]"
    }
  ]
}
