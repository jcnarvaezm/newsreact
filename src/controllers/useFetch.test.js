import React from 'react';
import { mount } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import NewsContainer from '../components/newsContainer/newsContainer';
import { create, act } from 'react-test-renderer';
import data from './data.json';
import { waitFor } from '@testing-library/dom';
import fetchMock from 'fetch-mock';

configure({
  adapter: new Adapter(),
});

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
  useEffect: jest.fn(),
}));

const API =
  'http://newsapi.org/v2/top-headlines?country=us&category=business&apiKey=b9c5c16a422d4b7b90accf49eed04cbc';

describe('Testing useFetch', () => {
  let wrapper;
  const setDataNews = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  const useStateMock: any = (initState: any) => [initState, setDataNews];
  useStateSpy.mockImplementation((dataNews) => [dataNews, setDataNews]);

  beforeEach(() => {
    jest.spyOn(React, 'useEffect').mockImplementation(() => {});
    global.fetch = fetch;
  });

  afterEach(() => {
    jest.clearAllMocks();
    fetchMock.restore();
  });

  it('calls setDataNews with 1', () => {
    jest.spyOn(React, 'useState').mockImplementation(useStateMock);
    wrapper = mount(<NewsContainer count={1} />);
    expect(setDataNews).toHaveBeenCalledTimes(0);
  });

  test('useEffect test', async (done) => {
    jest.spyOn(React, 'useEffect').mockImplementation();
    let component = create(<NewsContainer count={1} />);
    act(() => {
      component.update();
      expect(component.toJSON()).toMatchSnapshot();
      done();
    });
    await waitFor(() => expect(data).toHaveBeenCalledTimes(1));
  });
});
