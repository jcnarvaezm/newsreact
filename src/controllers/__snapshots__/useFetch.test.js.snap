// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing useFetch useEffect test 1`] = `
<div
  className="eui-panel eui-panel-default"
>
  <div
    className="eui-panel-body"
  >
    <h2>
      Loading...!!!
    </h2>
    <div
      className="eui-row"
    />
  </div>
</div>
`;
